fastlane_core/lib/fastlane_core/project.rb

```diff
@@ -301,13 +312,33 @@ def self.xcode_list_timeout
     end
 
     # @internal to module
-    # runs the specified command and kills it if timeouts
-    # @raises Timeout::Error if timeout is passed
+    def self.xcode_build_settings_timeout
+      (ENV['FASTLANE_XCODEBUILD_SETTINGS_TIMEOUT'] || 10).to_i
+    end
+
+    # @internal to module
+    def self.xcode_build_settings_retries
+      (ENV['FASTLANE_XCODEBUILD_SETTINGS_RETRIES'] || 3).to_i
+    end
+
+    # @internal to module
+    # runs the specified command and kills it if timeouts, optionally retries before killing
+    # @raises Timeout::Error if timeout is passed after all retry attempts
     # @returns the output
-    # Note: currently affected by fastlane/fastlane_core#102
-    def self.run_command(command, timeout: 0)
+    # Note: - currently affected by fastlane/fastlane_core#102
+    #       - retry feature added to solve fastlane#4059
+    def self.run_command(command, timeout: 0, retries: 0)
       require 'timeout'
-      @raw = Timeout.timeout(timeout) { `#{command}`.to_s }
+
+      tries = 1
+      begin
+        return Timeout.timeout(timeout) { `#{command}`.to_s }
```

Comment:
I will 👍 

---

I use Bundler to manage the tools we use (CocoaPods, Fastlane, and Danger), basically I just put the git branch I want to use for Fastlane. I was successfully using $4492 when I decided to create a simplified PR using the same implementation it did but totally compartmentalized to `project.rb` this is the progress of changes:

1. #4492, worked
2. #4492 applied to `project.rb`, worked
3. [early feedback](https://github.com/fastlane/fastlane/pull/5188#issuecomment-229654362) asked to use `Helper.backticks`, retries stopped working and the build hand just like Fastene in master
4. Went back to the original backticks in `project.rb` instead of `Helper.backticks`, and it worked again (literally nothing else changed)

All these I put to run on our CI which makes the problem happen more often (probably because of the many times the build runs).